### PR TITLE
workerutil: Add QueuedCount to store

### PR DESCRIPTION
**Pull request by efritz**
This PR adds a method to get the number of queued jobs to the internal/worker store. This will be used to determine the number of items in the queue and its growth rate over time for metrics collection and alerting.

This closes https://github.com/sourcegraph/sourcegraph/issues/14832.

### CODE REVIEW

File: internal/workerutil/dbworker/store/store.go [hunk 1]
```diff
@@ -216,6 +219,22 @@ func (s *store) Transact(ctx context.Context) (*store, error) {
 	return &store{Store: txBase, options: s.options, columnReplacer: s.columnReplacer}, nil
 }
 
+// QueuedCount returns the number of records in the queued state matching the given conditions.
+func (s *store) QueuedCount(ctx context.Context, conditions []*sqlf.Query) (int, error) {
+	count, _, err := basestore.ScanFirstInt(s.Query(ctx, s.formatQuery(
+		queuedCountQuery,
+		quote(s.options.ViewName),
```
Comment (eseliger) [inline internal/workerutil/dbworker/store/store.go:226]: The docblock states if ViewName is not set Table name is used, does that happen here?

**Review by eseliger — APPROVED** [review]

File: internal/workerutil/dbworker/store/store.go [hunk 2]
```diff
@@ -216,6 +219,22 @@ func (s *store) Transact(ctx context.Context) (*store, error) {
...
+		makeConditionSuffix(conditions),
+	)))
+
+	return count, err
+}
+
+const queuedCountQuery = `
+-- source: internal/workerutil/store.go:QueuedCount
+SELECT COUNT(*) FROM %s WHERE {state} = 'queued' %s
```
Comment (mrnugget) [inline internal/workerutil/dbworker/store/store.go:235]: _Technically_ an errored job is also queued if it hasn't reached `MaxNumRetries`, right?

**Review by mrnugget — APPROVED** [review]

File: internal/workerutil/dbworker/store/store.go (hunk 1 above)
Comment (efritz) [reply internal/workerutil/dbworker/store/store.go:226]: Yes, when the store is constructed we do

```
if options.ViewName == "" {
	options.ViewName = options.TableName
}
```

File: internal/workerutil/dbworker/store/store.go (hunk 2 above)
Comment (efritz) [reply internal/workerutil/dbworker/store/store.go:235]: Good catch!

**Pull request merged**